### ISSUE TITLE
xgboost4j-spark : Training the model via DStream (Streaming)

### ISSUE THREAD

**Issue by SundeepMehta** [dmlc/xgboost#2877]
Does xgboost4j supports DStream(Streaming Api) for training the model ? I don't see any api method which takes DStream as an input

Any advise on the same or pointers to get it implemented, I am looking to incremental train model as an when I receive the data via Stream.

**Comment by superbobry** [comment]
Hey @SundeepMehta, nope at the moment this is not implemented. I think it could be tricky to implement unless we move the distribution part to Spark and use C++ code only for updating/computing the gradients and split finding. Right now xgboost4j relies on Rabit for distribution, a custom AllReduce implementation.
